src/server_manager/scripts/get_packaging_opts.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -eu
+#
+# Copyright 2020 The Outline Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Collects common packaging options.  Meant to be called from the diffrent package_foo and 
+# release_foo scripts.
+# Usage from a packaging script:  
+#     src/server_manager/electron_app/package_$PLATFORM $0 $@
+
+readonly YARN_COMMAND="yarn do $(echo "${1}" | sed 's:.*\(src/server_manager/electron_app/.*\)_action.sh:\1:')"
+shift
+
+function usage () {
+  echo "Usage:" 1>&2
+  echo "${YARN_COMMAND} [-s stagingPercentage]" 1>&2
+  echo "  -s: The staged rollout percentage for this release.  Must be in the interval (0, 100].  Defaults to 100" 1>&2
+  echo "  -h: this help message" 1>&2
+  echo 1>&2
+  echo "Examples:" 1>&2
+  echo "Releases the beta of version 1.2.3 to 10% of users listening on the beta channel" 1>&2
+  echo '$ '"jq -r '.version' src/server_manager/package.json'" 1>&2
+  echo "1.2.3-beta" 1>&2
+  echo '$ '"${YARN_COMMAND} -s 10" 1>&2
+  exit 1
+}
+
+STAGING_PERCENTAGE=100
+while getopts s:? opt; do
+  case $opt in
+    s) STAGING_PERCENTAGE=$OPTARG ;;
+    *) usage ;;
+  esac
+done
+
+if ((STAGING_PERCENTAGE <= 0)) || ((STAGING_PERCENTAGE > 100)); then
+  echo "Staging percentage must be greater than 0 and no more than 100" 1>&2
+  exit 1
+fi
+
+# This breaks if we have more than one option.  We could, for example, use jq here to return JSON
+# and jq from the calling script to parse it to variables.
+echo "${STAGING_PERCENTAGE}"
```

Comment:
This doesn't quite work as you propose, since subprocesses get a copy of the parent process' environment.  I did some research, however, and found that scripts called with `source` run in the same process as the caller, and so can populate variables without them even needing to be exported to the environment at all!  Thanks for the motivation to find a better solution \^_\^

Done in  3ba6a3c

Reply:
Glad my suggestion helped 😅